psc-flink/src/main/java/com/pinterest/flink/streaming/connectors/psc/table/PscConnectorOptionsUtil.java

```diff
@@ -96,6 +101,57 @@ class PscConnectorOptionsUtil {
     private static final List<String> SCHEMA_REGISTRY_FORMATS =
             Arrays.asList(AVRO_CONFLUENT, DEBEZIUM_AVRO_CONFLUENT);
 
+    /**
+     * Validates if AUTO_GEN_UUID is allowed for the given property key and that app.prefix is provided.
+     * Only allows AUTO_GEN_UUID for psc.consumer.client.id, psc.consumer.group.id, and psc.producer.client.id keys.
+     * Throws ValidationException if AUTO_GEN_UUID is not allowed for the key or if app.prefix is missing/empty.
+     *
+     * @param subKey the property key (without the "properties." prefix)
+     * @param tableOptions the table options map to check for app.prefix
+     * @throws ValidationException if AUTO_GEN_UUID is not allowed for this key or app.prefix is invalid
+     */
+    public static void validateAutoGenUuidOptions(String subKey, Map<String, String> tableOptions) {
+        if (!(PscConfiguration.PSC_CONSUMER_CLIENT_ID.equals(subKey) || 
+              PscConfiguration.PSC_CONSUMER_GROUP_ID.equals(subKey) || 
+              PscConfiguration.PSC_PRODUCER_CLIENT_ID.equals(subKey))) {
+            throw new ValidationException(
+                    String.format("AUTO_GEN_UUID is not allowed for property '%s'. " +
+                    "AUTO_GEN_UUID is only supported for: %s, %s, %s", 
+                    subKey, 
+                    PscConfiguration.PSC_CONSUMER_CLIENT_ID,
+                    PscConfiguration.PSC_CONSUMER_GROUP_ID,
+                    PscConfiguration.PSC_PRODUCER_CLIENT_ID));
+        }
+        
+        // Validate that client.id.prefix is provided and non-empty when using AUTO_GEN_UUID
+        String clientIdPrefix = tableOptions.get("properties.client.id.prefix");
```

Comment:
nit: call the static variable here - `PscConnectorOptions.PROPS_CLIENT_ID_PREFIX.key()`